gensim/models/phrases.py

```diff
@@ -106,17 +206,23 @@ def __init__(self, sentences=None, min_count=5, threshold=10.0,
         `delimiter` is the glue character used to join collocation tokens, and
         should be a byte string (e.g. b'_').
 
+        `average_sentence_size`: expected number of tokens in average sentence.
+            max_vocab_size / average_sentence_size sentences is used to build python
+            dictionary per thread.
+            Lowering `average_sentence_size` will increase memory demands.
         """
         if min_count <= 0:
-            raise ValueError("min_count should be at least 1")
-
+            raise ValueError("min_count should be 1")
+        if min_count > 1:
+            logger.warning("min_count should be 1")
```

Comment:
Agreed with @gojomo -- we don't want to keep dead code around.

And in case of `Phrases`, we don't need the "old" behaviour either. The Phrases functionality is new and not widely used yet, so a clear release note saying things have changed seems enough here.